src/libsystemd/sd-device/sd-device.c

```diff
@@ -325,6 +325,10 @@ _public_ int sd_device_new_from_subsystem_sysname(sd_device **ret, const char *s
         if (access(syspath, F_OK) >= 0)
                 return sd_device_new_from_syspath(ret, syspath);
 
+        syspath = strjoina("/sys/firmware/", subsystem, "/", sysname);
```

Comment:
That's an interesting point, but my main focus here was keeping backwards compatibility, so I guess if one has to shadow the other, it's best to do things the way this patch suggests.

Also, it seems that nobody ever needed to access /sys/firmware for dmi (which has been around for a significant number of years), so I guess this won't really be a problem anytime soon.